src/qt/pivx/mnselectiondialog.cpp

```diff
@@ -111,7 +111,11 @@ void MnSelectionDialog::updateView()
     QFlags<Qt::ItemFlag> flgCheckbox = Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsUserCheckable;
     QFlags<Qt::ItemFlag> flgTristate = Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsUserCheckable | Qt::ItemIsTristate;
 
+    bool isLegacyObselete = mnModel->isLegacySystemObsolete();
     for (int i = 0; i < mnModel->rowCount(); ++i) {
+        uint8_t mnType = mnModel->index(i, MNModel::TYPE, QModelIndex()).data().toUInt();
+        if (mnType == MNViewType::LEGACY && isLegacyObselete) continue;
+        if (mnType == MNViewType::DMN_OPERATOR || mnType == MNViewType::DMN_OWNER) continue;
```

Comment:
shouldn't the type be treated as a bit flag? i.e. `if (mnType != MNViewType::LEGACY && (mnType | MNViewType::DMN_VOTER) == 0) continue`

Reply:
yeah.. it should.